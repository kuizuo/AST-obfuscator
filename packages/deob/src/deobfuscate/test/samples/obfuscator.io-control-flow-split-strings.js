function _0x22ee(_0xff4276, _0x22ee3a) {
  const _0x20f9e3 = _0xff42()
  _0x22ee = function (_0x42500c, _0x3edd5d) {
    _0x42500c = _0x42500c - 0x0
    const _0x14e3e1 = _0x20f9e3[_0x42500c]
    return _0x14e3e1
  }
  return _0x22ee(_0xff4276, _0x22ee3a)
}
function _0xff42() {
  const _0x234db9 = [
    '3|4|0|',
    '2|1',
    'wsdHS',
    'Hello\x20',
    'World!',
    'split',
    'log',
  ]
  _0xff42 = function () {
    return _0x234db9
  }
  return _0xff42()
}
function _0x225c6a() {
  const _0x4b8108 = _0x22ee
  const _0x335abf = {}
  _0x335abf.cmKxi = _0x4b8108(0x0) + _0x4b8108(0x1)
  _0x335abf[_0x4b8108(0x2)] = _0x4b8108(0x3) + _0x4b8108(0x4)
  const _0x2789be = _0x335abf
  const _0x4f432d = _0x2789be.cmKxi[_0x4b8108(0x5)]('|')
  let _0x5114c4 = 0x0
  while ([]) {
    switch (_0x4f432d[_0x5114c4++]) {
      case '0':
        console[_0x4b8108(0x6)](_0x2789be[_0x4b8108(0x2)])
        continue
      case '1':
        console[_0x4b8108(0x6)](_0x2789be[_0x4b8108(0x2)])
        continue
      case '2':
        console[_0x4b8108(0x6)](_0x2789be[_0x4b8108(0x2)])
        continue
      case '3':
        console[_0x4b8108(0x6)](_0x2789be[_0x4b8108(0x2)])
        continue
      case '4':
        console.log(_0x2789be[_0x4b8108(0x2)])
        continue
    }
    break
  }
}
_0x225c6a()
