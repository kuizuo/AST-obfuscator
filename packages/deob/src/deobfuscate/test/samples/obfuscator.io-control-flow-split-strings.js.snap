function _0x225c6a() {
  console.log("Hello World!");
  console.log("Hello World!");
  console.log("Hello World!");
  console.log("Hello World!");
  console.log("Hello World!");
}
_0x225c6a();